### ISSUE TITLE
tcp-server

### ISSUE THREAD

**Issue by Bettman66** [juha-h/baresip-studio#251]
Is there a possibility to install the ctrl-tcp plugin from baresip. An open user interface would be available to control baresip studio.

**Comment by juha-h** [comment]
Walter Zengel writes:

> Is there a possibility to install the ctrl-tcp plugin from baresip. An
> open user interface would be available to control baresip studio.

Might be.  We'll check when I have time.

**Comment by juha-h** [comment]
baresip Android app build from this repository is GUI app that does not include baresip menu module and thus not register any of its commands.  baresip app calls baresip lib API calls based on user GUI actions, handles resulting events, and displays their effects on the GUI.

Mapping of 60+ JSON messages and responses to them to baresip GUI actions would be a very big task.  Or are you suggesting that there would be no GUI actions associated with them?  If so, then a new project could be started to implement such an Android app.

**Comment by Bettman66** [comment]
Yes you are right, I didn't want to use the frontend. I had imagined controlling the app with TCP sockets, so any programming language for Android can be used to control baresip. Creating a service for the lib that can be controlled with TCP sockets would be great, but it's beyond my programming skills.

**Comment by juha-h** [comment]
I think the app would still need GUI for settings and account creation/configuration, backup/restore, etc.  Also, if video is going to be supported, the GUI would need to provide window for video display.

Anyway, this would be a new app.

**Comment by juha-h** [comment]
I'll close this issue now since it is not related to baresip-studio based apps.
